tests/integration/utils/mock_XHR.js

```diff
@@ -0,0 +1,74 @@
+import parseXML from "./parseXML.js"
+
+/*
+* Partial mock of XMLHttpRequest
+* @param {function} GetContent
+* @param {Object} Mock
+*/
+export default function mockXHR(GetContent, Mock) {
```

Comment:
Why adding the getContent argument? Will it change? It already respond to a specific interface ({ data, type, size }) which seems to be specifically tailored for mockXHR.

Reply:
Why capitalizing the name? This should be reserved for classes. You do not have to create a const based on it after, as arguments values are expected to not be changed (that would be poor style).

Reply:
There's also other points that annoys me:
  - This is not a _true_ mock/stub of an XHR as its name hints at, only one which has an effect on __our__ implementation of it, and only for manifest+segments requests.
  - The name of the function also does not make it clear that it's only for a single mock, which now are for single contents only
  - This implementation should be temporary, as the real reason for this code is a bug in the ``sinon.js`` library. We should add comments to make it clear and add/follow our issue on the right repo